src/components/Layout/LanguageSelector.tsx

```diff
@@ -85,29 +96,34 @@ export const LanguageSelector = () => {
     [activePage.languages, languageVersions],
   );
 
-  const queryParams = new URLSearchParams(location.search);
+  const queryParams = useMemo(() => new URLSearchParams(location.search), [location.search]);
   const langParam = queryParams.get('lang');
 
+  useEffect(() => {
+    if (langParam && !options.some((option) => option.label === langParam)) {
+      queryParams.delete('lang');
+      navigate(`${location.pathname}?${queryParams.toString()}`);
```

Comment:
[nitpick] I don't think we need to keep the "invalid" language in the user's history...

~~~suggestion
      navigate(`${location.pathname}?${queryParams.toString()}`, {replace: true});
~~~